share/util/pcd_const.F90

```diff
@@ -0,0 +1,59 @@
+! WARNING! DO NOT EDIT THIS FILE!
+! This file was generated automatically from pcd.yaml
+! For more information, visit https://github.com/e3sm-project/PhysicalConstantsDictionary
+
+module pcd
+    implicit none
+
+    !define double precision kind
+    integer, parameter :: dp = selected_real_kind(12)
+
+    ! pcd.yaml version and institution information
+    character(len=*), parameter :: pcdversion = "0.0.2"
+    character(len=*), parameter :: pcdinstitution = "E3SM Project"
+
+    !mathematical constants
+    real(dp), parameter :: pi = 3.141592653589793_dp
+    real(dp), parameter :: e = 2.718281828459045_dp
+    real(dp), parameter :: em_gamma = 0.5772156649015329_dp
+    real(dp), parameter :: radian = 57.29577951308232_dp
+    real(dp), parameter :: degree = 0.017453292519943295_dp
+    real(dp), parameter :: square_root_of_2 = 1.4142135623730951_dp
+    real(dp), parameter :: square_root_of_3 = 1.7320508075688772_dp
+
+    !universal_physical constants
+    real(dp), parameter :: speed_of_light_in_vacuum = 299792458_dp
+    real(dp), parameter :: newtonian_gravitation_constant = 6.6743e-11_dp
+    real(dp), parameter :: standard_acceleration_of_gravity = 9.80665_dp
+    real(dp), parameter :: standard_atmosphere = 101325_dp
+    real(dp), parameter :: avogadro_constant = 6.02214076e+23_dp
+    real(dp), parameter :: boltzmann_constant = 1.380649e-23_dp
+    real(dp), parameter :: stefan_boltzmann_constant = 5.670374419e-08_dp
+    real(dp), parameter :: planck_constant = 6.62607015e-34_dp
+    real(dp), parameter :: molar_gas_constant = 8.314462618_dp
+    real(dp), parameter :: molar_volume_of_ideal_gas = 0.02271095464_dp
+
+    !earth_physical constants
+    real(dp), parameter :: geocentric_gravitational_constant = 3986005E+08_dp
+    real(dp), parameter :: semimajor_axis = 6378137_dp
+    real(dp), parameter :: dynamic_form_factor = 108263E-08_dp
+    real(dp), parameter :: angular_velocity = 7292115E-11_dp
+    real(dp), parameter :: semiminor_axis = 6356752.3141_dp
+    real(dp), parameter :: flattening = 0.00335281068118_dp
+    real(dp), parameter :: reciprocal_flattening = 298.257222101_dp
+    real(dp), parameter :: mean_radius = 6371008.7714_dp
+    real(dp), parameter :: radius_of_sphere_of_same_surface = 6371007.181_dp
+    real(dp), parameter :: radius_of_sphere_of_same_volume = 6371000.79_dp
+
+    !solar constants
+    real(dp), parameter :: total_solar_irradiance = 1360.8_dp
+
+    !water constants
+    real(dp), parameter :: water_molar_mass = 18.015268_dp
+    real(dp), parameter :: water_specific_gas_constant = 0.46151805_dp
+    real(dp), parameter :: water_triple_point_temperature = 273.16_dp
+    real(dp), parameter :: water_triple_point_pressure = 611.655_dp
+    real(dp), parameter :: liquid_water_triple_point_density = 999.793_dp
```

Comment:
Consider adding a constant for the density of pure ice (at some temperature), if there is one.